### ISSUE TITLE
CSRF issue when logging in

### ISSUE THREAD

**Issue by WoLpH** [rtfd/readthedocs.org#1268]
When I login to readthedocs I always get a CSRF error, since it logs me in anyway it doesn't matter too much (potential security issue there?). But it's still annoying and quite ugly since it's one of the standard Django errors:

![screen shot 2015-05-15 at 14 27 48](https://cloud.githubusercontent.com/assets/270571/7652645/f20c4882-fb0e-11e4-9172-182b9561609b.png)

Some proper styling would help here, but I've got a feeling there's some security issue going on as well. If it really gets a CSRF error it shouldn't log you in, or it shouldn't check if there's nothing worrysome to check ;)

**Comment by ericholscher** [comment]
How are you logging in? If this was a common issue, I imagine we'd have more bugs open about it.

Also, if it was really a security issue, you shouldn't report it on a public issue tracker. All security issues should be reported privately. I don't believe it is though, and is likely something wrong with your browser or environment.

**Comment by WoLpH** [comment]
Well, it might be a small security issue. Nothing remotely exploitable.

Anyway, just logging in normally with chrome and ghostery and adblock
installed. It's probably related to missing referrer headers which is not
really an issue generally but annoying nonetheless.

As for the security issue, apparently you get logged in regardless of csrf
issues. Don't think it's exploitable but it's weird. By default django
would block all user input if csrf validation fails. Somehow it still logs
me in... That's curious.

**Comment by gregmuellegger** [comment]
I tried to reproduce it, I used a fresh install of chromium and installed these two plugins:
https://chrome.google.com/webstore/detail/ghostery/mlomiejdfkolichcflejclcbmpeaniij
https://chrome.google.com/webstore/detail/adblock/gighmmpiobklfepjocnamgkkbiglidom

Then I went to readthedocs.org and tried logging in. That worked fine. I then logged out again and logged in again, but this time I messed with the CSRF token in the form by hand. Then I also saw the message about the CSRF verification failed. However I was not logged in afterwards.

Is there any more information you can provide us with? It's not reproduceable so far on our end :(

**Comment by WoLpH** [comment]
I've just tried again and it apears that I am not able to reproduce it anymore. My guess is that it was a certain combination of Chrome, Ghostery and Adblock which has since been resolved in one of these projects.

The CSRF error itself is still reproducible by blocking referrer headers (several client-side firewalls do this) but it's more of a Django bug that missing/empty headers also give the error.

To override the styling the csrf failure view can be used: https://docs.djangoproject.com/en/1.8/ref/settings/#csrf-failure-view
